install/first_google_boot.sh

```diff
@@ -215,21 +219,42 @@ function extract_spinnaker_kube_credentials() {
     kube_zone=$MY_ZONE
   fi
 
+  if clear_metadata_to_file "kube_config" $config_path; then
+    echo "Attempting to write kube_config to $config_path..."
+    if [[ -s $config_path ]]; then
+      chmod 400 $config_path
+      chown spinnaker $config_path
+      echo "Successfully wrote kube_config to $config_path"
+
+      write_default_value "SPINNAKER_KUBERNETES_ENABLED" "true"
+    else
+      echo "Failed to write kube_config to $config_path"
+      rm $config_path
+    fi
+
+    # Safely return since we have attempted to configure Kubernetes with the
+    # provided metadata.
+    return
+  else
+    clear_instance_metadata "kube_config"
+  fi
+
   if [ -n "$kube_cluster" ]; then
     echo "Downloading credentials for cluster $kube_cluster in project $kube_project in zone $kube_zone..."
     export KUBECONFIG=$config_path
     gcloud config set container/use_client_certificate true
     gcloud container clusters get-credentials $kube_cluster --zone $kube_zone --project $kube_project
```

Comment:
FYI, the gcloud call doesnt work to reach into another project because you dont have the credentials for it.
If the google_credentials passed in are also for that project, then you can register them into an account with gcloud and then use that --account flag here and I think it would work. To give the credentials to gcloud, see https://cloud.google.com/sdk/gcloud/reference/auth/activate-service-account